src/components/encounter/encounter-form.component.tsx

```diff
@@ -465,26 +465,30 @@ const EncounterForm: React.FC<EncounterFormProps> = ({
         });
       }
       // handle attachments
-      try {
-        const attachmentsResponse = await Promise.all(
-          EncounterFormManager.saveAttachments(fields, savedEncounter, abortController),
-        );
-        if (attachmentsResponse?.length) {
-          showSnackbar({
-            title: t('attachmentsSaved', 'Attachment(s) saved successfully'),
-            kind: 'success',
-            isLowContrast: true,
+
+      if (EncounterFormManager.evaluatedAttachmentFields(fields)?.length) {
+        try {
+          const attachmentsResponse = await Promise.all(
+            EncounterFormManager.saveAttachments(fields, savedEncounter, abortController),
```

Comment:
Can you just make sure that if there are zero attachments this function returns an empty value for `attachmentsResponse` and implement the necessary filters internally?

Reply:
You mean something like:

```ts
const evaluatedFields = EncounterFormManager.evaluatedAttachmentFields(fields);
if (evaluatedFields?.length) {
  try {
    const attachmentsResponse = await Promise.all(
      EncounterFormManager.saveAttachments(fields, savedEncounter, abortController),
    );
    if (attachmentsResponse?.length) {
      showSnackbar({
        title: t('attachmentsSaved', 'Attachment(s) saved successfully'),
        kind: 'success',
        isLowContrast: true,
      });
    }
  } catch (error) {
    const errorMessages = extractErrorMessagesFromResponse(error);
    return Promise.reject({
      title: t('errorSavingAttachments', 'Error saving attachment(s)'),
      subtitle: errorMessages.join(', '),
      kind: 'error',
      isLowContrast: false,
    });
  }
} else {
  return [];
}

```

Reply:
I think the logic can be handled in the `EncounterFormManager.saveAttachments` function, 

```
  static saveAttachments(fields: FormField[], encounter: OpenmrsEncounter, abortController: AbortController) {
    const complexFields = fields?.filter((field) => field?.questionOptions.rendering === 'file');
    return complexFields.map((field) => {
      const patientUuid = typeof encounter?.patient === 'string' ? encounter?.patient : encounter?.patient?.uuid;
      return saveAttachment(
        patientUuid,
        field,
        field?.questionOptions.concept,
        new Date().toISOString(),
        encounter?.uuid,
        abortController,
      );
    });
  }
```

Somewhere in here, before` return saveAttachment(`

Reply:
Thanks for the clarification @CynthiaKamau